src/react/hooks/useQuery.ts

```diff
@@ -83,8 +83,17 @@ class InternalState<TData, TVariables> {
   constructor(
     public readonly client: ReturnType<typeof useApolloClient>,
     public readonly query: DocumentNode | TypedDocumentNode<TData, TVariables>,
+    previous?: InternalState<TData, TVariables>,
   ) {
     verifyDocumentType(query, DocumentType.Query);
+
+    // Reuse previousData from previous InternalState (if any) to provide
+    // continuity of previousData even if/when the query or client changes.
+    const previousResult = previous && previous.result;
+    const previousData = previousResult && previousResult.data;
```

Comment:
As this is Typescript, I guess we can safely use optional-chaining as I expect this to be transpiled if needed during build:
```suggestion
    const previousData = previous?.result?.data;
```

Reply:
This completely makes sense and I've never actually checked the transpiled output. Thanks for the insightful explanation!